From 0a052a88bf96542e0bb7585ccbfd39e681a32e3d Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 4 Dec 2019 19:06:54 -0800
Subject: [PATCH] Delete compiler options not available in release versions of
 clang _yet_

* CREL feature is not supported on the current clang toolchain version,
  which causes build to fail with the following error:

| clang++: error: unsupported argument '--crel' to option '-Wa,'
| clang++: error: unsupported argument '--allow-experimental-crel'
  to option '-Wa,'

Upstream-Status: Inappropriate [ clang/master already supports them ]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Randy MacLeod <Randy.MacLeod@windriver.com>
Signed-off-by: Ariel D'Alessandro <ariel.dalessandro@collabora.com>

---
 build/config/compiler/BUILD.gn | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index b0e9e04bf3..46ed84d73c 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -614,13 +614,6 @@ config("compiler") {
     } else {
       cflags += [ "-ffp-contract=off" ]
     }
-
-    # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF
-    # (excluding toolchains that use an older version of LLVM).
-    if (is_linux && !llvm_android_mainline &&
-        default_toolchain != "//build/toolchain/cros:target") {
-      cflags += [ "-Wa,--crel,--allow-experimental-crel" ]
-    }
   }

   # C11/C++11 compiler flags setup.
@@ -1934,6 +1927,11 @@ config("default_warnings") {
       }

       cflags += [
+        # Disable unknown warnings, since system clang may not have
+        # backported newer warning yet unlike internal clang that
+        # chromium uses.
+        "-Wno-unknown-warning-option",
+
         # TODO(crbug.com/330524456): -Wcast-function-type is under -Wextra now.
         "-Wno-cast-function-type",

